plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt

```diff
@@ -407,6 +408,11 @@ class KotlinGenerator : SharedCodegen() {
         getHeadersToIgnore().forEach { headerName ->
             ignoreHeaderParameter(headerName, codegenOperation)
         }
+
+        // Let's remove the leading
+        if (!basePath.isNullOrBlank()) {
```

Comment:
I did check few introductory docs and [videos](https://www.youtube.com/watch?v=KIAoQbAu3eA&feature=youtu.be&t=32m19s) related to this issue and the proposed approach looks the safest one to use in this case.

One not is related to the value of `base_path`.

If the spec specify `base_path` as `/<something_not_suffixed_by_slash>` (ie. `/v3` in the petstore APIs) how is retrofit going to build the final url?

Reply:
Given that we don't build the Retrofit base url (but is delegated to the user of the generator), I don't think this is an issue that we can address right now.